be/src/vec/columns/column_array.cpp

```diff
@@ -311,7 +311,7 @@ void ColumnArray::insert_range_from(const IColumn & src, size_t start, size_t le
         cur_offsets.assign(src_offsets.begin(), src_offsets.begin() + length);
     } else {
         size_t old_size = cur_offsets.size();
-        size_t prev_max_offset = old_size ? cur_offsets.back() : 0;
```

Comment:
？cur_offsets is empty still get `cur_offesets.back()`

Reply:
Yes, `-1` here is ok due to the `Offsets` (PaddedPODArray<Uint32>) is always padded zero left here. If you refer to the method `offset_at` in class `ColumnArray`, you will notice that the offset of the beginning element (position 0) is at the position -1 in offsets.

Reply:
Added a comment to explain this.